### DISCUSSION TITLE
Tuya Smart Plugs states drops to zero frequently

### DISCUSSION THREAD

**Discussion by Re4zOon** [Koenkk/zigbee2mqtt#16076] [Devices]
Hi,

Dont know if anyone come accross a similar issue, but...
A picture tells better stories:
![image](https://user-images.githubusercontent.com/9249934/211529708-5c075a46-2e2c-4f5b-adda-802fafa82c11.png)

All those vertical lines are datapoints for "0" watt.
My question is: can I somehow average these datapoints, so I dont go mad?
It can be a new sensor for all I care, I just want a nice graph :)

More info:
I have several of these exact plugs, but only some of these do them.
There are no updates to them, everything is running the same FW, etc.
Model: TS011F_plug_1

I see the mention on the page about the polling, but its not recognized as _3 so I dont have that option.

**Comment by kzajac83** [comment]
The same issue for  in my" TS011F_plug_1" and for my colleges using the same device in HA. In meantime update FW via OTA but nothing change. 
Please note using ZHA have the same issue so problem probalby can by in socket or Zigbee concentrator but also  try on two different  device butissue still  the same (however chip in both the same CC2652P). On the other hand other data in the same time like voltage is valid then communcation problaby is not a problem. 


 Because few hoursago  change ZHA -> Z2M then my history is short but see the  issue:


![image](https://user-images.githubusercontent.com/50861081/231571257-22c55456-0ab2-491a-acb9-f6276c4e749e.png)

**Comment by Dash** [comment]
I too am seeing these drops to zero
![image](https://github.com/Koenkk/zigbee2mqtt/assets/129838/d67d59df-a933-4210-a090-5af19d03c9bb)
I've only got one of these sockets at the moment, and is generally unbranded otherwise. I assume this is a firmware issue (OTA v.192), but no idea who to feed that back to.

Power and Current both show the same problem and the drop to 0 occurs at the same time.
![image](https://github.com/Koenkk/zigbee2mqtt/assets/129838/6bd99a14-9bbe-44b3-9ba1-af83a1f7db63)

**Comment by kzajac83** [comment]
Lastly buy some zigbee socket and after connect to Z2M is visible as TS011F but both sockets have others manufacturer codes:

![Screenshot_2023-05-11-13-41-54-479_io homeassistant companion android](https://github.com/Koenkk/zigbee2mqtt/assets/50861081/6c7ced08-0905-4889-8e93-07425557afa8)

So have two sockets registerd as TS011F and one report zero ramdomly (see two post above) and new one working fine:

![Screenshot_2023-05-11-13-36-24-983_io homeassistant companion android](https://github.com/Koenkk/zigbee2mqtt/assets/50861081/2cce262b-6ba2-44ce-b105-ec94a5fca7b1)

Both OTA / FW version is 192.  

Badly socket manufacter is _TZ3000_hdopuwv6 and working correctly _TZ3000_gjnozsaz

**Reply by Dash** [reply]
Interesting, mine is "_TZ3000_cehuw1lw", but no way of identifying that before buying.

**Comment by Re4zOon** [comment]
For me it doesn't really matter why it happens, its obvious its not an issue with Z2M.
However, as its a "common" problem, we need a workaround of sorts.
As I said, even a custom entity, that averages the last 10 datapoints or something is fine.

Since I made my original post I learned some about HA and how to use templates and such, so possibly I will solve this for myself and paste the config here.

**Comment by Dash** [comment]
This has been fixed with issue #16076. If you're getting the same thing on a different device, then it seems adding a comment to that issue is the way.

The current list of devices with exceptions is: '_TZ3000_gvn91tmx', '_TZ3000_amdymr7l', '_TZ3000_typdpbpg', '_TZ3000_hdopuwv6', '_TZ3000_b28wrpvx', '_TZ3000_1h2x4akh', '_TZ3000_cehuw1lw'

For these devices, if the value goes to 0, it'll hold off for 5 seconds before sending a zero value through.
https://github.com/Koenkk/zigbee-herdsman-converters/blob/75d28701a7b321ab95c4137a4ce753ff329a2178/src/devices/tuya.ts#L445C4-L445C4

**Comment by katlandreth** [comment]
I'm not sure if I'm supposed to make a new issue or add to this one. My smart plug (_TZ3008_esl4rq9g) seems to be having the same issue but it doesn't look like it's listed in the device exceptions. Is there any way to confirm it's the same problem and get this plug added to the list? Thanks!

**Reply by andrei-lazarov** [reply]
The "prevent 0 values" fix is now applied to all TS011F plugs. Are you on the latest Z2M?

**Reply by dan73800** [reply]
Hi
I am joining this thread because I have the same issue with some plugs TS011F reconnized as '_TZ3000_amdymr7l'.
They often go to 0 as mentionned earlier.
I do not really understand (sorry) what I have to do to prevent this. I actually use release 2.12.0 of Z2M and it does not seem to be fixed.
Do I have something to do with the link : https://github.com/Koenkk/zigbee-herdsman-converters/blob/75d28701a7b321ab95c4137a4ce753ff329a2178/src/devices/tuya.ts#L445C4-L445C4
posted by @Dash ?
Thanks for your answer
Dan

**Reply by andrei-lazarov** [reply]
You don't have to do anything. It's already part of Z2M.

**The plugs are faulty and they give random 0 values.**
Z2M should wait a few seconds before publishing the 0 values, effectively giving the plugs time to correct themselves with a new reporting.

You can click the Reconfigure button to make sure everything is set up correctly.
Otherwise you can try this firmware update (download the raw file and upload in Z2M UI):
https://github.com/Koenkk/zigbee-OTA/blob/master/images/Tuya/1662545193-oem_zg_tl8258_plug_OTA_3.0.0.bin
It's officially for `_TZ3000_ufv0gpfo`, but users report is works on other `TS011F` plugs too. It might fix the issue.

**Reply by dan73800** [reply]
Thanks for your help.
I have successfuly used reconfigure button, but nothing special happens.
I can't ugrade with the firmware in your url. When I use it with OTA I have this error : 
`[2026-07-02 18:42:15] [31merror[39m: 	zh:controller:device: Failed to parse OTA image from 'https://github.com/Koenkk/zigbee-OTA/blob/master/images/Tuya/1662545193-oem_zg_tl8258_plug_OTA_3.0.0.bin' for 0xa4c138d0b0587578, aborting (Not a valid OTA file)
`

**Reply by andrei-lazarov** [reply]
I said to download the raw file and upload in Z2M UI

<img width="316" height="227" alt="image" src="https://github.com/user-attachments/assets/662fe2fe-d56a-4a81-81a0-9659e98bb5fa" />

<img width="400" height="463" alt="image" src="https://github.com/user-attachments/assets/0c24fe36-9068-478c-85a2-888b41de491a" />


This is the raw direct link
https://github.com/Koenkk/zigbee-OTA/raw/refs/heads/master/images/Tuya/1662545193-oem_zg_tl8258_plug_OTA_3.0.0.bin

**Reply by dan73800** [reply]
Hi
It is exactly what I have done, but it does not want to update

<img width="379" height="497" alt="image" src="https://github.com/user-attachments/assets/8fd7fe82-3e90-4eb2-93b6-c0666623c051" />

Always the same error :
[03/07/2026 14:13:56] z2m: Update of 'Prise Pompe' failed (No image currently available)
And I have downloaded the RAW file. So do not understand why I can't update it

I have even tried with the URL directly in the field, but same trouble

<img width="371" height="432" alt="image" src="https://github.com/user-attachments/assets/4d3f3898-4f06-4791-8bf3-f07583b6c5c0" />

**Reply by dan73800** [reply]
I have found what was the trouble.
My plug was already with the last firmware. 192 means 3.0.0 ;)
I have tried to downgrade with 1.1.2 (82) and the downgrade works, but still the same trouble with power = 0
I bring it back to the last firmware 3.0.0 and it also works, but does not solve my problem.
I suppose I have to change plugs to have right values.
Thanks for help
Dan
